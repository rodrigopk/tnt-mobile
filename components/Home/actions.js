export const DATA_AVAILABLE = 'MESSAGE_RECEIVED';

//Import the sample data
import Data from '../../hello.json';

export const fetchHelloMessage = () => {
    return (dispatch) => {

        setTimeout(() => {
            var message  = Data.message;
            dispatch({type: DATA_AVAILABLE, message: message});
        }, 2000);

    };
};

export default { fetchHelloMessage };
