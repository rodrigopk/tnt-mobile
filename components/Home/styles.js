import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  activityIndicatorContainer:{
    backgroundColor: "#fff",
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
  },
  messageContainer: {
    flex:1,
    backgroundColor: '#F5F5F5',
    paddingTop:20,
    justifyContent: 'center',
    alignItems: 'center',
  },
  message:{
    fontSize: 21,
  },
});

export default styles;
