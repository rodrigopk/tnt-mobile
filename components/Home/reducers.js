import { combineReducers } from 'redux';

import { MESSAGE_RECEIVED } from "./actions"

let initialState = {
  message: '',
  title: '',
  loading:true,
};

const homeReducer = (state = initialState, action) => {
    switch (action.type) {
        case MESSAGE_RECEIVED:
            state = Object.assign(
              {},
              state,
              {
                title: action.title,
                message: action.message,
                loading:false,
              });
            return state;
        default:
            return state;
    }
};

export default homeReducer;
