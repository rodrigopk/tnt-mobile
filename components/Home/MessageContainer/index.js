import React, { Component } from 'react';
import { View, Text } from 'react-native';

import styles from './styles';

const MessageContainer = ({ title, message }) => (
  <View style={styles.container}>
    <Text style={styles.title}>
      { title }
    </Text>
    <Text style={styles.message}>
      { message }
    </Text>
  </View>
);

export default MessageContainer;
