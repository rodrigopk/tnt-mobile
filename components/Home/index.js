import React, { Component } from 'react';
import {
  View,
  Text,
  ActivityIndicator,
} from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { fetchHelloMessage } from './actions';
import styles from './styles';

class Home extends Component {
  componentDidMount() {
    this.props.getMessage();
  }

  render() {
    if (this.props.loading) {
      return (
          <View style={styles.activityIndicatorContainer}>
              <ActivityIndicator
                  animating={true}
                  style={[{height: 80}]}
                  size="small"
              />
          </View>
      );
    } else {
      return (
        <View style={styles.messageContainer}>
          <Text style={styles.message}>
            { this.props.message }
          </Text>
        </View>
      );
    }
  }
};

const mapStateToProps = state => ({
  loading: state.home.loading,
  message: state.home.message,
});

const mapDispatchToProps = dispatch => ({
  getMessage: () =>  dispatch(fetchHelloMessage()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Home);
