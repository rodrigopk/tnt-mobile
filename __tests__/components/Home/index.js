import 'react-native';
import 'isomorphic-fetch';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'

import Home from '../../../components/Home/index.js';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const loadingState = {
  home: {
    message: '',
    title: '',
    loading: true
  }
};

const finishedLoadingState = {
  home: {
    message: 'Hello from the backend!',
    title: 'Greetings',
    loading: false
  }
};

describe('Testing Home component', () => {
  it('renders loading state', () => {
    const wrapper = shallow(
      <Home />,
      { context: { store: mockStore(loadingState) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('renders loaded state', () => {
    const wrapper = shallow(
      <Home />,
      { context: { store: mockStore(finishedLoadingState) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
