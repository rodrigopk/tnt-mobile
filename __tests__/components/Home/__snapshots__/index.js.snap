// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home component renders loaded state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    getMessage={[Function]}
    loading={false}
    message="Hello from the backend!"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "message": "Hello from the backend!",
      "title": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "message": "Hello from the backend!",
        "title": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;

exports[`Testing Home component renders loading state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    getMessage={[Function]}
    loading={true}
    message=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
        style={
          Array [
            Object {
              "height": 80,
            },
          ]
        }
      />,
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animating": true,
        "color": "#999999",
        "hidesWhenStopped": true,
        "size": "small",
        "style": Array [
          Object {
            "height": 80,
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="small"
          style={
            Array [
              Object {
                "height": 80,
              },
            ]
          }
        />,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animating": true,
          "color": "#999999",
          "hidesWhenStopped": true,
          "size": "small",
          "style": Array [
            Object {
              "height": 80,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
